draft-irtf-cfrg-bbs-signatures.md

```diff
@@ -584,21 +727,63 @@ ABORT if:
 
 Procedure:
 
-1.  (Q_1, MsgGenerators) = create_generators(L+1, PK)
-2.  (H_1, ..., H_L) = MsgGenerators
-3.  (H_i1, ..., H_iR) = (MsgGenerators[i1], ..., MsgGenerators[iR])
-4.  (H_j1, ..., H_jU) = (MsgGenerators[j1], ..., MsgGenerators[jU])
-5.  domain = calculate_domain(PK, Q_1, (H_1, ..., H_L), header)
-6.  D = P1 + Q_1 * domain + H_i1 * msg_i1 + ... + H_iR * msg_iR
-7.  T =  Abar * r2^ + Bbar * r3^ + H_j1 * m^_j1 + ... +  H_jU * m^_jU
-8.  T = T + D * c
-9.  cv = calculate_challenge(Abar, Bbar, T, (i1, ..., iR),
-                             (msg_i1, ..., msg_iR), domain, ph)
-10. if c != cv, return INVALID
-11. if e(Abar, W) * e(Bbar, -BP2) != Identity_GT, return INVALID
-12. return VALID
+1. (Q_1, MsgGenerators) = create_generators(L+1, PK)
+2. (H_1, ..., H_L) = MsgGenerators
+3. (H_i1, ..., H_iR) = (MsgGenerators[i1], ..., MsgGenerators[iR])
+4. (H_j1, ..., H_jU) = (MsgGenerators[j1], ..., MsgGenerators[jU])
+5. domain = calculate_domain(PK, Q_1, (H_1, ..., H_L), header)
+6. D = P1 + Q_1 * domain + H_i1 * msg_i1 + ... + H_iR * msg_iR
+7. T =  Abar * r2^ + Bbar * r3^ + H_j1 * m^_j1 + ... +  H_jU * m^_jU
+8. T = T + D * c
+9. return (Abar, Bbar, T, domain)
+```
+
+### Challenge Calculation
+
+This operation calculates the challenge scalar value, used during [ProofGen](#proof-generation-proofgen) and [ProofVerify](#proof-verification-proofverify), as part of the Fiat-Shamir heuristic, for making the proof protocol non-interactive (in a interactive sating, the challenge would be a random value supplied by the verifier).
+
+This operation makes use of the `serialize` function, defined in [Section 4.6.1](#serialize).
+
+```
+challenge = ChallengeCalculate(init_res, i_array, msg_array, ph)
```

Comment:
Maybe ProofChallengeCalculate for precision?

Reply:
Done. Thanks!